source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -564,7 +564,7 @@ bool M2MConnectionHandlerPimpl::init_socket()
     _running = true;
     palSocketType_t socket_type = PAL_SOCK_DGRAM;
     palStatus_t status;
-    palSocketDomain_t domain;
+    palSocketDomain_t domain = PAL_AF_UNSPEC;
```

Comment:
As a side note, the change in "socket domain = PAL_AF_UNSPEC" rings a bell, there was a bug in mbed-os 5.2.2 on something similar socket address family bein set to unspecified. Is that covered on the CI tests? The change you did looks correct as is, but I am just wondering does that case get testing coverage. The domain could also be set to unspec on a new "} else " branch at line 583, so we could even verify from the coverage report if that case is really being tested.